CMakeLists.txt

```diff
@@ -55,11 +59,19 @@ SET(CMAKE_BUILD_COLOR_MAKEFILE          ON)
 SET(CMAKE_CXX_FLAGS "-std=gnu++1y ${COMMON_WARNING_FLAGS} -Wnon-virtual-dtor ${MACHINE_FLAGS} ${GLIBC_COMPATIBILITY_COMPILE_FLAGS}")
 SET(CMAKE_CXX_FLAGS_RELEASE             "-O3 -DNDEBUG")
 SET(CMAKE_CXX_FLAGS_RELWITHDEBINFO      "-O3 -g")
+
 SET(CMAKE_C_FLAGS                       "${COMMON_WARNING_FLAGS} ${MACHINE_FLAGS} ${GLIBC_COMPATIBILITY_COMPILE_FLAGS}")
 SET(CMAKE_C_FLAGS_RELEASE               "-O3 -DNDEBUG")
 SET(CMAKE_C_FLAGS_RELWITHDEBINFO        "-O3 -g")
 SET(CMAKE_EXE_LINKER_FLAGS              "-static-libgcc -static-libstdc++ ${GLIBC_COMPATIBILITY_LINK_FLAGS}")
 
+SET(CMAKE_CXX_FLAGS_ASAN                "-O3 -g -fsanitize=address -fno-omit-frame-pointer")
+SET(CMAKE_CXX_FLAGS_TSAN                "-O3 -g -fsanitize=thread -fno-omit-frame-pointer")
```

Comment:
BTW, for thread sanitizer it is not enough, because you need to pass this flag both for compiler and linker, and also remove -static-libgcc, -static-libstdc++
And also, thread sanitizer just don't work with libstdc++ (emits false positives, tested with gcc 6).
You need either:
- rebuild libstdc++ with TSan;
- rebuild with clang;
- use libc++ instead of libstdc++.

So, this option is broken right now. Need to fix or remove.